src/jit/morph.cpp

```diff
@@ -5036,6 +5037,15 @@ GenTreePtr Compiler::fgMorphMultiregStructArg(GenTreePtr arg, fgArgTabEntryPtr f
             noway_assert(elemCount <= 4);
 #endif
 
+            if (varDsc->lvGcLayout == nullptr)
+            {
+                // I am not sure whether we should initialize lvGcLayout or not
```

Comment:
I am not sure about this code, perhaps we should not go through the loop bellow if lvGcLayout hasn't been initialized

Reply:
Yeah, I don't think you need this. Instead you might assert that if the gc layout is null, the type does not contain gc references (see for instance the first few lines of `impNormStructType`).

Reply:
Fixed